pkg/repository/v1/trigger.go

```diff
@@ -103,11 +103,25 @@ func newTriggerRepository(s *sharedRepository) TriggerRepository {
 	}
 }
 
-func (r *TriggerRepositoryImpl) TriggerFromEvents(ctx context.Context, tenantId string, opts []EventTriggerOpts) ([]*sqlcv1.V1Task, []*DAGWithData, error) {
+type EventDataForOlap struct {
+	EventKey                string
+	EventId                 string
+	TaskExternalId          string
+	EventAdditionalMetadata []byte
+	EventPayload            []byte
+}
+
+type TriggerFromEventsResult struct {
+	Tasks  []*sqlcv1.V1Task
+	Dags   []*DAGWithData
+	Events []*EventDataForOlap
```

Comment:
I think making this something like `map[string][]*Run` will make life much easier here (will need to create a type for `Run`). You don't need to worry about the `n^2` problem above or removing duplicate events in the OLAP controller.

Reply:
There's a bunch of downstream stuff that uses the `Tasks` and `Dags` from here - do you think we should return the `map[string][]*Run` and then refactor the downstream things to use `Run` instead of e.g. `V1Task` and `DagWithData`?

Reply:
ahhh nvm I think I'm understanding, one sec

Reply:
took another stab at this but also just returning a list, but still prevents the N^2 thing afaict - lmk if you think that's fine!